include/tester/db/VTDatabase.hpp

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include "Database.hpp"
+
+class VTDatabase : Database
+{
+public:
+    VTDatabase(const std::string &domain):
+        Database(domain)
+    {     
+    }
+
+    ~VTDatabase() = default;
+
+    void save(const bsoncxx::document::value &document)
+    {
+        Database::save(COLLECTION_NAME ,document);
+    }
+
+    bsoncxx::document::value getAllHeaders()
+    {
+        mongocxx::collection collection = _database[COLLECTION_NAME];
+        mongocxx::cursor cursor = collection.find({});
+
+        bsoncxx::builder::stream::document result{};
+        bsoncxx::builder::stream::array headers{};
+
+        for (auto&& doc : cursor) 
+        {
+            bsoncxx::builder::stream::document header_builder{};
+            header_builder << "_id" << doc["_id"].get_oid().value.to_string();
+
+            if(doc["creation-time"])
+            {
+                header_builder << "creation-time" << doc["creation-time"].get_string().value.to_string();
+            }
+
+            headers << header_builder;
+        }
+
+        result << "headers" << headers;
+
+        return result.extract();
+    }
+
+    std::vector<bsoncxx::document::value> getAll()
+    {
+        return Database::getAll(COLLECTION_NAME);
```

Comment:
I don't think you need to use `Database::` as a prefix in these methods. what you can do is mark the relevant methods in the parent class `Database` to use the `protected` modifier.

Reply:
And since you've made `Database` a 'generic' class - how about renaming this method to `getTestLogs`?

Reply:
Done

Reply:
> I don't think you need to use `Database::` as a prefix in these methods. what you can do is mark the relevant methods in the parent class `Database` to use the `protected` modifier.

About that. If I have the same name for functions in both classes, then I need `Database::`. If I have another name for the functions I can do it without `Database::` and just call the function.

Reply:
Even if `Database::getAll(std::string collectionName)` is `protected`?

Correct me if I'm wrong, but the way I expect is to be able to call the `Database` class method `getAll(COLLECTION_NAME)` inside of the `VTDatabase` class method `getAll()`. That way, you don't expose the `getAll(std::string collectionName)` method from `Database` and just use `getAll()` from `VTDatabase`

And that way you get rid of that `Database::` prefix.

Reply:
Done